src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -1202,6 +1206,16 @@ bool FancyZones::IsNewWorkArea(GUID virtualDesktopId, HMONITOR monitor) noexcept
     return true;
 }
 
+void FancyZones::LoadWorkAreas()
+{
+    const auto& fancyZonesData = JSONHelpers::FancyZonesDataInstance();
+    const auto usedVirtualDesktops = fancyZonesData.GetVirtualDeslktopIds();
+    for (auto id : usedVirtualDesktops)
+    {
+        m_processedWorkAreas[id] = std::vector<HMONITOR>();

```

Comment:
You are filling up `m_processedWorkAreas` and relying on `FancyZones::RegisterVirtualDesktopUpdates` to clean up persisted data for those work areas that no longer exist ?

Reply:
I was also wondering about this.

Reply:
Yes, I thought why not to use ready mechanism instead of creating something new. Was it bad idea?

Reply:
Not necessarily a bad idea, but RegisterVirtualDesktopUpdates is called whenever something changes in VirtualDesktops registry key, new virtual desktop added, deleted etc. (although I’ve noticed that is actually called more often). My point is that You don’t need to wait for something to change, to handle this, You can ask for all virtual desktop ids right away (from that registry). I’m working on similar stuff right now (check #2339 and functions I added in VirtualDesktopUtils namespace), so we can merge our work maybe.

Reply:
@vldmr11080 thank you for the tip!
I noticed that RegisterVirtualDesktopUpdates is always called on app start, so there is no need to wait changes. But on the other hand at least regarding code readability it would be better to explicitly request virtual desktop ids and update settings file. 
I took a look at your PR and `GetVirtualDekstopIds` can help in my case. 
Since I'll wait with merge until 0.18 I think I could rebase after you merge that PR and update my PR then.